src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/PropertyMetadata.cs

```diff
@@ -62,7 +62,10 @@ public PropertyMetadata(IProperty property, Type dbContextType)
 
             IsEnum = property.ClrType.GetTypeInfo().IsEnum;
             IsReadOnly = property.IsReadOnlyBeforeSave;
-            IsAutoGenerated = property.IsStoreGeneratedAlways;
+
+            IsAutoGenerated = ((property.RequiresValueGenerator
+                || property.IsStoreGeneratedAlways)
+                && (property.ValueGenerated != ValueGenerated.Never));
```

Comment:
> Values can be generated when entities start being tracked, and values can be generated/changed when an entity is saved to the database. However, in neither case is a value necessarily generated

@ajcvickers So based on the `IProperty` what's the best way to determine whether a property's value can be set by the application or not? More specifically while creating entities.